app/services/articles/attributes.rb

```diff
@@ -0,0 +1,44 @@
+module Articles
+  class Attributes
+    ATTRIBUTES = %i[archived body_markdown canonical_url description
+                    edited_at main_image organization_id user_id published
+                    title video_thumbnail_url].freeze
+
+    attr_reader :attributes, :article_user
+
+    def initialize(attributes, article_user)
+      @attributes = attributes
+      @article_user = article_user
+    end
+
+    def for_update(update_edited_at: false)
+      hash = {}
```

Comment:
I may be misreading this but I think the first 4 lines basically boil down to something like this (`hash` contains only the key-value-pairs from `attributes` for the keys listed in `ATTRIBUTES`):

```ruby
hash = attributes.slice(*ATTRIBUTES)
```